### ISSUE TITLE
Index out of range error

### ISSUE THREAD

**Issue by working12** [UKPLab/sentence-transformers#461]
This is the error I got. I have checked the data, there is no empty field or anything. Where am I going wrong?
```
---------------------------------------------------------------------------
IndexError                                Traceback (most recent call last)
<ipython-input-21-6c7c3accc56a> in <module>()
      1 binary_acc_evaluator = evaluation.BinaryClassificationEvaluator(dev_sentences1, dev_sentences2, dev_labels, show_progress_bar=True)
----> 2 binary_acc_evaluator(model, epoch=0, steps=0,output_path=model_save_path)

9 frames
/usr/local/lib/python3.6/dist-packages/sentence_transformers/evaluation/BinaryClassificationEvaluator.py in __call__(self, model, output_path, epoch, steps)
     80         logging.info("Binary Accuracy Evaluation of the model on " + self.name + " dataset" + out_txt)
     81         embeddings1 = model.encode(self.sentences1, batch_size=self.batch_size,
---> 82                                    show_progress_bar=self.show_progress_bar, convert_to_numpy=True)
     83         embeddings2 = model.encode(self.sentences2, batch_size=self.batch_size,
     84                                    show_progress_bar=self.show_progress_bar, convert_to_numpy=True)

/usr/local/lib/python3.6/dist-packages/sentence_transformers/SentenceTransformer.py in encode(self, sentences, batch_size, show_progress_bar, output_value, convert_to_numpy, convert_to_tensor, is_pretokenized, device, num_workers)
    166             iterator = tqdm(inp_dataloader, desc="Batches")
    167 
--> 168         for features in iterator:
    169             for feature_name in features:
    170                 features[feature_name] = features[feature_name].to(device)

/usr/local/lib/python3.6/dist-packages/tqdm/notebook.py in __iter__(self, *args, **kwargs)
    213     def __iter__(self, *args, **kwargs):
    214         try:
--> 215             for obj in super(tqdm_notebook, self).__iter__(*args, **kwargs):
    216                 # return super(tqdm...) will not catch exception
    217                 yield obj

/usr/local/lib/python3.6/dist-packages/tqdm/std.py in __iter__(self)
   1102                 fp_write=getattr(self.fp, 'write', sys.stderr.write))
   1103 
-> 1104         for obj in iterable:
   1105             yield obj
   1106             # Update and possibly print the progressbar.

/usr/local/lib/python3.6/dist-packages/torch/utils/data/dataloader.py in __next__(self)
    361 
    362     def __next__(self):
--> 363         data = self._next_data()
    364         self._num_yielded += 1
    365         if self._dataset_kind == _DatasetKind.Iterable and \

/usr/local/lib/python3.6/dist-packages/torch/utils/data/dataloader.py in _next_data(self)
    401     def _next_data(self):
    402         index = self._next_index()  # may raise StopIteration
--> 403         data = self._dataset_fetcher.fetch(index)  # may raise StopIteration
    404         if self._pin_memory:
    405             data = _utils.pin_memory.pin_memory(data)

/usr/local/lib/python3.6/dist-packages/torch/utils/data/_utils/fetch.py in fetch(self, possibly_batched_index)
     45         else:
     46             data = self.dataset[possibly_batched_index]
---> 47         return self.collate_fn(data)

/usr/local/lib/python3.6/dist-packages/sentence_transformers/SentenceTransformer.py in smart_batching_collate_text_only(self, batch)
    404         """
    405 
--> 406         max_seq_len = max([self._text_length(text) for text in batch])
    407         feature_lists = {}
    408 

/usr/local/lib/python3.6/dist-packages/sentence_transformers/SentenceTransformer.py in <listcomp>(.0)
    404         """
    405 
--> 406         max_seq_len = max([self._text_length(text) for text in batch])
    407         feature_lists = {}
    408 

/usr/local/lib/python3.6/dist-packages/sentence_transformers/SentenceTransformer.py in _text_length(self, text)
    426         (representing several text inputs to the model).
    427         """
--> 428         if isinstance(text[0], int):
    429             return len(text)
    430         else:

IndexError: list index out of range
```

This is how i am creating the evaluation data
```
dev_sentences1 = []
dev_sentences2 = []
dev_labels = []
for index, row in devset.iterrows():
        dev_sentences1.append(row['question1'])
        dev_sentences2.append(row['question2'])
        dev_labels.append(int(row['is_duplicate']))


binary_acc_evaluator = evaluation.BinaryClassificationEvaluator(dev_sentences1, dev_sentences2, dev_labels, show_progress_bar=True)
binary_acc_evaluator(model, epoch=0, steps=0,output_path=model_save_path)
```

**Comment by nreimers** [comment]
Can you try the most recent version from github and try again?

You can install it like this (when you are in the checked-out folder):
```
pip install -e .
```

It appears that some input of you does not produce any token by the tokenizer.

The most recent version in the repository should fix it.

**Comment by working12** [comment]
Thanks for the reply. I have installed it like this in google colab. `!pip install sentence_transformers`. So I should do this instead `!pip install -e sentence_transformers`?

**Comment by working12** [comment]
I have installed it like this. `!pip install git+https://github.com/UKPLab/sentence-transformers`. But still, I get this. 

```
---------------------------------------------------------------------------
IndexError                                Traceback (most recent call last)
<ipython-input-12-6c7c3accc56a> in <module>()
      1 binary_acc_evaluator = evaluation.BinaryClassificationEvaluator(dev_sentences1, dev_sentences2, dev_labels, show_progress_bar=True)
----> 2 binary_acc_evaluator(model, epoch=0, steps=0,output_path=model_save_path)

9 frames
/content/sentence-transformers/sentence_transformers/evaluation/BinaryClassificationEvaluator.py in __call__(self, model, output_path, epoch, steps)
     80         logging.info("Binary Accuracy Evaluation of the model on " + self.name + " dataset" + out_txt)
     81         embeddings1 = model.encode(self.sentences1, batch_size=self.batch_size,
---> 82                                    show_progress_bar=self.show_progress_bar, convert_to_numpy=True)
     83         embeddings2 = model.encode(self.sentences2, batch_size=self.batch_size,
     84                                    show_progress_bar=self.show_progress_bar, convert_to_numpy=True)

/content/sentence-transformers/sentence_transformers/SentenceTransformer.py in encode(self, sentences, batch_size, show_progress_bar, output_value, convert_to_numpy, convert_to_tensor, is_pretokenized, device, num_workers)
    166             iterator = tqdm(inp_dataloader, desc="Batches")
    167 
--> 168         for features in iterator:
    169             for feature_name in features:
    170                 features[feature_name] = features[feature_name].to(device)

/usr/local/lib/python3.6/dist-packages/tqdm/notebook.py in __iter__(self, *args, **kwargs)
    213     def __iter__(self, *args, **kwargs):
    214         try:
--> 215             for obj in super(tqdm_notebook, self).__iter__(*args, **kwargs):
    216                 # return super(tqdm...) will not catch exception
    217                 yield obj

/usr/local/lib/python3.6/dist-packages/tqdm/std.py in __iter__(self)
   1102                 fp_write=getattr(self.fp, 'write', sys.stderr.write))
   1103 
-> 1104         for obj in iterable:
   1105             yield obj
   1106             # Update and possibly print the progressbar.

/usr/local/lib/python3.6/dist-packages/torch/utils/data/dataloader.py in __next__(self)
    361 
    362     def __next__(self):
--> 363         data = self._next_data()
    364         self._num_yielded += 1
    365         if self._dataset_kind == _DatasetKind.Iterable and \

/usr/local/lib/python3.6/dist-packages/torch/utils/data/dataloader.py in _next_data(self)
    401     def _next_data(self):
    402         index = self._next_index()  # may raise StopIteration
--> 403         data = self._dataset_fetcher.fetch(index)  # may raise StopIteration
    404         if self._pin_memory:
    405             data = _utils.pin_memory.pin_memory(data)

/usr/local/lib/python3.6/dist-packages/torch/utils/data/_utils/fetch.py in fetch(self, possibly_batched_index)
     45         else:
     46             data = self.dataset[possibly_batched_index]
---> 47         return self.collate_fn(data)

/content/sentence-transformers/sentence_transformers/SentenceTransformer.py in smart_batching_collate_text_only(self, batch)
    408 
    409         for text in batch:
--> 410             sentence_features = self.get_sentence_features(text, max_seq_len)
    411             for feature_name in sentence_features:
    412                 if feature_name not in feature_lists:

/content/sentence-transformers/sentence_transformers/SentenceTransformer.py in get_sentence_features(self, *features)
    313 
    314     def get_sentence_features(self, *features):
--> 315         return self._first_module().get_sentence_features(*features)
    316 
    317     def get_sentence_embedding_dimension(self):

/content/sentence-transformers/sentence_transformers/models/Transformer.py in get_sentence_features(self, tokens, pad_seq_length)
     74         pad_seq_length = min(pad_seq_length, self.max_seq_length, self.auto_model.config.max_position_embeddings-3) + 3 #Add space for special tokens
     75 
---> 76         if isinstance(tokens[0], int):
     77             return self.tokenizer.prepare_for_model(tokens, max_length=pad_seq_length, padding='max_length', return_tensors='pt', truncation=True, prepend_batch_axis=True)
     78         else:

IndexError: list index out of range
```

**Comment by nreimers** [comment]
Could you post a minimal working example, that lead to this error.

**Comment by working12** [comment]
```
import pandas as pd
from datetime import datetime 
import os
import torch
import torch.nn as nn
import numpy as np
from sentence_transformers import losses, util
from sentence_transformers import SentenceTransformer, models, SentencesDataset, evaluation
from sentence_transformers.readers import InputExample
from torch.utils.data import DataLoader


word_embedding_model = models.Transformer('bert-base-uncased', max_seq_length=256)
pooling_model = models.Pooling(word_embedding_model.get_word_embedding_dimension())
dense_model = models.Dense(in_features=pooling_model.get_sentence_embedding_dimension(), out_features=256, activation_function=nn.Tanh())
model = SentenceTransformer(modules=[word_embedding_model, pooling_model, dense_model],device='cuda')

num_epochs = 10
train_batch_size = 8
#As distance metric, we use cosine distance (cosine_distance = 1-cosine_similarity)
distance_metric = losses.SiameseDistanceMetric.COSINE_DISTANCE
margin = 0.5
model_save_path = 'output/training_OnlineConstrativeLoss-'+datetime.now().strftime("%Y-%m-%d_%H-%M-%S")

os.makedirs(model_save_path, exist_ok=True)
trainset = pd.read_csv('trainset.csv')
devset = pd.read_csv('devset.csv')
trainset = trainset.dropna()
devset = devset.dropna()
# train_examples = [InputExample(texts=['My first sentence', 'My second sentence'], label=0.8)
train_samples = []
for index, row in trainset.iterrows():
    #if row['question1']==' '
    #print(row['question1'],row['question2'],row['id'])
    sample = InputExample(texts=[row['question1'], row['question2']], label=int(row['is_duplicate']))
    train_samples.append(sample)
dev_sentences1 = []
dev_sentences2 = []
dev_labels = []
for index, row in devset.iterrows():
        #print(row['question1'],row['question2'],row['is_duplicate'])
        dev_sentences1.append(row['question1'])
        dev_sentences2.append(row['question2'])
        dev_labels.append(int(row['is_duplicate']))

binary_acc_evaluator = evaluation.BinaryClassificationEvaluator(dev_sentences1, dev_sentences2, dev_labels, show_progress_bar=True)
binary_acc_evaluator(model, epoch=0, steps=0,output_path=model_save_path)
```

This the code I am trying to run. @nreimers

**Comment by nreimers** [comment]
Could you please post a **running** example, i.e. an example that I could also run and that does not depend on any (unknown) external files.

**Comment by working12** [comment]
@nreimers Actually there is nothing else there. I just have two data files that I am sharing here. https://drive.google.com/drive/folders/16QQBK89SMgC9waUcINjhUSM_Wy0t4S93?usp=sharing

Those were prepared by me earlier.

**Comment by nreimers** [comment]
The issue was connected to empty strings.

It should be fixed in the most recent version on pypi (0.3.7.1).

Install it with:
```
pip install -U sentence-transformers
```

**Comment by working12** [comment]
Thanks @nreimers
